app/models/order.rb

```diff
@@ -125,6 +125,21 @@ def init_dates
     self
   end
 
+  def self.map_with_own_group_orders(scope, ordergroup, page = nil, limit = nil)
+    orders = scope.includes(:supplier)
+    orders = orders.page(page).per(limit) unless limit.nil?
+    own_orders = scope.includes(:group_orders).where(group_orders: {ordergroup_id: ordergroup.id}, id: orders.map {|o| o.id})
+    own_orders_hash = Hash[own_orders.collect {|o| [o.id, o]}]
+    orders.map do |order|
+      group_order = own_orders_hash.has_key?(order.id) ? own_orders_hash[order.id].group_orders.first : nil
+
+      {
+        order: order,
+        group_order: group_order
+      }
```

Comment:
Yes, but `group_order` is `nil`, if the current user hasn't placed one for this `order`:

In those lists we want to show orders "enriched" with group_orders - SQL-wise this is a `LEFT OUTER JOIN`. In Rails this could simply be written by `Order.includes(:group_orders).where({group_orders: {ordergroup_id: ordergroup.id}})`, but while this will internally do the LEFT OUTER JOIN, it will still just return orders where a matching ordergroup could be found.

I could not find a way to bypass this behavior without deeply interfering with ActiveRecord.

Also even if it were possible, we'd have orders that all have exactly one or zero elements in their CollectionProxy, which I personally find misleading.

I agree that the hash isn't so nice. I'll change it into an `attr_accessor :cached_ordergroups_group_order` on the `Order` model - let me know, if you prefer that too or have a different proposal!